promql/value.go

```diff
@@ -168,6 +168,23 @@ func (p HPoint) MarshalJSON() ([]byte, error) {
 	return json.Marshal([...]interface{}{float64(p.T) / 1000, h})
 }
 
+// histogramSize returns the number of samples representing the histogram.
+// The total size is calculated considering the histogram timestamp (p.T - 8 bytes),
+// and then a number of bytes in the histogram.
+// This sum is divided by 16, as samples are 16 bytes.
+func (p HPoint) histogramSize() int {
+	return (p.H.Size() + 8) / 16
+}
+
+// totalHistogramSize returns the total number of samples in the given slice of histograms.
+func totalHistogramSize(histograms []HPoint) int {
```

Comment:
I would call this `totalHPointSize` as it includes the timestamp and the pointer to the actual histogram.